Dockerfile-recursor

```diff
@@ -0,0 +1,45 @@
+# USAGE
+
+#   docker build --build-arg MAKEFLAGS=-j8 -t recursor -f docker/Dockerfile-recursor .
+#   docker run -p 1053:53 -p 1053:53/udp -ti --rm recursor
+#   dig a www.example.com @0 -p 1053
+
+# our chosen base image
+FROM debian:10
+
+# import everything - this could be pdns.git OR a recursor tarball!
+COPY . /source
+
+# TODO: make sure /source looks roughly the same from git or tar
+
+# TODO: ccache
+
+# devscripts gives us mk-build-deps (and a lot of other stuff)
+RUN apt-get update && apt-get -y dist-upgrade && apt-get install -y  --no-install-recommends devscripts equivs && apt-get clean
+
+# TODO: control file is not in tarballs at all right now
+RUN mk-build-deps -i -t 'apt-get -y -o Debug::pkgProblemResolver=yes --no-install-recommends' /source/builder-support/debian/recursor/debian-buster/control && apt-get clean
+RUN apt-get -y install protobuf-compiler && apt-get clean
+
+# build and install (TODO: before we hit this line, rearrange /source structure if we are coming from a tarball)
+WORKDIR /source/pdns/recursordist
+
+ARG MAKEFLAGS=
+ENV MAKEFLAGS ${MAKEFLAGS:--j2}
+
+#RUN touch pdns_recursor.1 rec_control.1 # avoid installing pandoc
```

Comment:
Do we still avoid installing 1 GB of pandoc in some other way?

Reply:
The comment should say 'avoid having to install pandoc'. pandoc does not currently end up in the images (but for various other reasons, many of which I'm sure I will avoid by doing multistage builds, the images are around 4GB right now).

Reply:
It is possible I get away with this because `COPY ... /source` is taking my unclean host tree that already has built manpages, will check.

Reply:
> It is possible I get away with this because `COPY ... /source` is taking my unclean host tree that already has built manpages, will check.

Confirmed!

Reply:
Hehe, definitely add a `.dockerignore` file to prevent shipping all your local build output to Docker.

Reply:
Just adding the Dockerfiles to .dockerignore has done wonders for build times at least!

Reply:
We now avoid pandoc by being explicit about our make targets.